tools/wake-format/emitter.cpp

```diff
@@ -409,9 +431,25 @@ static std::vector<CSTElement> collect_right_binary(CSTElement collect_over, CST
   CSTElement right = op;
   right.nextSiblingNode();
 
-  if (!(op.id() == CST_OP && op.firstChildElement().id() == collect_over.id() &&
-        op.firstChildElement().fragment().segment().str() ==
-            collect_over.fragment().segment().str())) {
+  // Stop collecting if we don't see an OP
+  if (op.id() != CST_OP) {
+    return {node};
+  }
+
+  // Stop collecting if the OP doesn't match both token and string contents
+  // Ex: OP_OR != OP_AND
+  // Ex: OP_OR("|") != OP_OR("||")
+  // Ex: OP_OR("||") == OP_OR("||")
+  bool matches = false;
+  for (const auto& i : collect_over) {
+    if (op.firstChildElement().id() == i.first &&
+        op.firstChildElement().fragment().segment().str() == i.second) {
+      matches = true;
+      break;
+    }
+  }
```

Comment:
Is this significant enough to DRY out?  I don't see any differences from the other block at a glance.

Reply:
Yeah its the same, I just didn't have a good enough name for it